File: src/main/java/com/nugurang/graphql/mutation/UserMutation.java
package com.nugurang.graphql.mutation;

import com.nugurang.dto.UserDto;
import com.nugurang.dto.UserInputDto;
import com.nugurang.service.UserService;
import graphql.kickstart.tools.GraphQLMutationResolver;
import org.springframework.stereotype.Service;

@Service
public class UserMutation implements GraphQLMutationResolver {
    private final UserService userService;

    public UserDto createCurrentUser(UserInputDto userInputDto) {
        return userService.createUser(userInputDto).toDto();
    }

    public UserDto updateCurrentUser(UserInputDto userInputDto) {
        return userService.updateCurrentUser(userInputDto).toDto();
    }

    public Long deleteUser(Long userId) {
        return userService.deleteUser(userId);
    }

    public Long deleteCurrentUser() {
        return userService.deleteCurrentUser();
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public UserMutation(final UserService userService) {
        this.userService = userService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/mutation/UserMutation.kt
package com.nugurang.graphql.mutation

import com.nugurang.dto.UserDto
import com.nugurang.dto.UserInputDto
import com.nugurang.mapper.UserMapper
import com.nugurang.service.UserService
import graphql.kickstart.tools.GraphQLMutationResolver
import org.springframework.stereotype.Service

@Service
class UserMutation(
    private val userService: UserService,
    private val userMapper: UserMapper
) : GraphQLMutationResolver {

    fun createCurrentUser(userInputDto: UserInputDto): UserDto {
        return userService.createUser(userInputDto).let(userMapper::toDto)
    }

    fun updateCurrentUser(userInputDto: UserInputDto): UserDto {
        return userService.updateCurrentUser(userInputDto).let(userMapper::toDto)
    }

    fun deleteUser(userId: Long): Long {
        return userService.deleteUser(userId)
    }

    fun deleteCurrentUser(): Long {
        return userService.deleteCurrentUser()
    }
}